src/Workspaces/CSharp/Portable/Recommendations/CSharpRecommendationServiceRunner.cs

```diff
@@ -451,6 +451,14 @@ private RecommendedSymbols GetSymbolsOffOfConditionalReceiver(ExpressionSyntax o
                 ? symbols.WhereAsArray(s => !(s.IsStatic || s is ITypeSymbol))
                 : (abstractsOnly ? symbols.WhereAsArray(s => s.IsAbstract) : symbols);
 
+
+            //If container type is "ref struct" then we should exclude methods from object and ValueType that are not overriden,
+            //because calling them produces a compiler error due to unallowed boxing. See https://github.com/dotnet/roslyn/issues/35178
+            if (containerType is not null && containerType.IsRefLikeType)
+            {
+                namedSymbols = namedSymbols.RemoveAll(s => s.ContainingType.SpecialType == SpecialType.System_Object || s.ContainingType.SpecialType == SpecialType.System_ValueType);
```

Comment:
```suggestion
                namedSymbols = namedSymbols.RemoveAll(s => s.ContainingType.SpecialType is SpecialType.System_Object or SpecialType.System_ValueType);
```

Reply:
Wow, I did'n actually know that enums can be compared using pattern matching